src/main/java/org/terasology/physicalstats/system/PhysicalStatsSystem.java

```diff
@@ -60,20 +60,32 @@ public void initialise() {
             if (clientEntity.hasComponent(HealthComponent.class)) {
                 HealthComponent h = clientEntity.getComponent(HealthComponent.class);
                 PhysicalStatsComponent p = clientEntity.getComponent(PhysicalStatsComponent.class);
-
-                // Ensure that the health is set to the appropriate percentage of the appropriate maximum health
-                if (h.maxHealth != p.constitution * 10) {
-                    float hP = (float) h.currentHealth / (float) h.maxHealth;
-                    h.maxHealth = p.constitution * 10;
-                    h.currentHealth = (int) Math.floor(h.maxHealth * hP);
-                    clientEntity.saveComponent(h);
-                }
+                updateHealth(clientEntity, h, p);
             }
         }
 
         super.initialise();
     }
 
+    /**
+     * Sets the health and maximum health values as components of the PhysicalStats constitution stat.
+     * @param e The entity (i.e. player) to update
+     * @param h The health stats of the player
+     * @param p The physical stats of the player
+     * @return True if changed, false if unchanged
+     */
+    public boolean updateHealth(EntityRef e, HealthComponent h, PhysicalStatsComponent p)
```

Comment:
You might be able to move the healthComponent inside the method, instead of taking it from the entity before passing it as a param since you have the entity as a param as well

Reply:
That also holds for the `PhysicalStatsComponent`, doesn't it? Making this a method with given signature at least allows to document the abbreviated argument names 🙄 

Honestly, I would appreciate more descriptive names, but with the doc this is fine (for now).